src/main/java/org/elasticsearch/index/query/CommonTermsQueryBuilder.java

```diff
@@ -42,58 +55,104 @@
  * execution times significantly if applicable.
  * <p>
  */
-public class CommonTermsQueryBuilder extends QueryBuilder implements BoostableQueryBuilder<CommonTermsQueryBuilder> {
+public class CommonTermsQueryBuilder extends QueryBuilder<CommonTermsQueryBuilder> implements BoostableQueryBuilder<CommonTermsQueryBuilder> {
 
     public static final String NAME = "common";
 
+    public static final float DEFAULT_CUTOFF_FREQ = 0.01f;
+
+    public static final Operator DEFAULT_HIGH_FREQ_OCCUR = Operator.OR;
+
+    public static final Operator DEFAULT_LOW_FREQ_OCCUR = Operator.OR;
+
+    public static final boolean DEFAULT_DISABLE_COORD = true;
+
     public static enum Operator {
-        OR, AND
+        OR, AND;
+
+        public Occur toMustOrShouldClause() {
+            return this.equals(Operator.AND) ? Occur.MUST : Occur.SHOULD;
+        }
+
+        public static Operator readFrom(StreamInput in, Operator defaultOperator) throws IOException {
+            Integer operator = in.readOptionalVInt();
+            if (operator == null) {
+                return defaultOperator;
+            } 
+            return operator == 1 ? AND : OR;
+        }
+
+        public static void writeTo(StreamOutput out, Operator operator, Operator defaultOperator) throws IOException {
+            if (operator == null || operator == defaultOperator) {
+                out.writeBoolean(false);
```

Comment:
I find it scary that we rely on internal behaviour of readOptionalVInt here, also looks like the following write is not optional right? I would rather try to have readFrom and writeTo here more aligned (although currently they would work, they are hard to read)